File: ruby/lib/sorpresa_ircasilla.rb
module Civitas
    class SorpresaIrCasilla < Sorpresa
        public_class_method :new
        def initialize(texto, tablero, valor)
            super(texto)
            @tablero=tablero
            @valor=valor
        end

        def aplicarAJugador(actual, todos)
            if jugadorCorrecto(actual, todos)
                informe(actual, todos)
                tirada = @tablero.calcularTirada(todos[actual].getNumCasillaActual, @valor)
                nueva_pos = @tablero.nuevaPosicion(todos[actual].getNumCasillaActual, tirada)
                todos[actual].moverACasilla(nueva_pos)
                @tablero.getCasilla(nueva_pos).recibeJugador(actual, todos)
            end
        end
    end
end

File: Civitas_gui/src/civitas/SorpresaIrCasilla.java
package civitas;

import java.util.ArrayList;

public class SorpresaIrCasilla extends Sorpresa {
    private Tablero tablero;
    private int valor;

    SorpresaIrCasilla(String texto, Tablero tablero, int valor) {
        super(texto);
        this.tablero = tablero;
        this.valor = valor;
    }

    public void aplicarAJugador(int actual, ArrayList<Jugador> todos) {
        if (jugadorCorrecto(actual, todos)){
            informe(actual,todos);
            int tirada = tablero.calcularTirada(todos.get(actual).getNumCasillaActual(), valor);
            int numCasilla = tablero.nuevaPosicion(todos.get(actual).getNumCasillaActual(), tirada);
            todos.get(actual).moverACasilla(numCasilla);
            tablero.getCasilla(valor).recibeJugador(actual, todos);
        }
    }
}